test.sh

```diff
@@ -7,52 +7,131 @@ GREEN="\e[32m"
 MAGENTA="\e[95m"
 RESET="\e[0m"
 
+# This function will be called when running this script
+function main(){
+    # This function will be called when running this script
+    run_tests_in_development;
+    run_tests_in_production;
+}
 
-# development or production?
-if [[ $(ls -l docker-compose.yml) == *"production.yml"* ]]; then
+function run_tests_in_development() {
+    echo -e "${MAGENTA}"
+    echo -e "=== Development mode ==="
+    echo -e "${RESET}"
+    WEBAPP_CONTAINER="osmaxxwebappdev"
+    CELERY_CONTAINER="osmaxxcelerydev"
+    DB_CONTAINER="osmaxxdatabasedev"
+    COMPOSE_FILE="compose-development.yml"
+    run_tests;
+}
+
+function run_tests_in_production() {
     echo -e "${MAGENTA}"
     echo -e "=== Production mode ==="
     echo -e "${RESET}"
     WEBAPP_CONTAINER="osmaxxwebapp"
     CELERY_CONTAINER="osmaxxcelery"
-else
-    echo -e "${MAGENTA}=== Develpment mode ==="
-    echo -e "${RESET}"
-    WEBAPP_CONTAINER="osmaxxwebappdev"
-    CELERY_CONTAINER="osmaxxcelerydev"
-fi
+    DB_CONTAINER="osmaxxdatabase"
+    COMPOSE_FILE="compose-production.yml"
+    run_tests;
+}
+
+function reset_containers() {
+    echo -e "resetting all containers";
+    dcompose stop -t 0 &> test.log;
+    dcompose rm -f &> test.log;
+    dcompose build &> test.log;
+}
 
+function reset_container() {
+    echo -e "resetting container ${CONTAINER_TO_BE_RESETTED};"
```

Comment:
It is an environment variable, which is set, but can not be accessed outside the script, since it is not exported. But if by global you mean 'available outside the scope of the `reset_container` function', yes. But it is not, in a programming sense, a global variable, but it is not a parameter either. TBH, I prefer this style, but don't care enough to blow it up to a _principles_ discussion :wink: .